documentation/api/io.strimzi.api.kafka.model.authentication.KafkaClientAuthenticationCustom.adoc

```diff
@@ -0,0 +1,90 @@
+To configure custom authentication, set the `type` property to `custom`.
+
+Custom authentication allows for any type of kafka supported authentication to be used.
+
+.Example Custom authentication for a bespoke OAuth setup
+[source,yaml,subs="attributes+"]
+----
+spec:
+  kafka:
+    config:
+      principal.builder.class: SimplePrincipal.class
+    listeners:
+      - name: oauth-bespoke
+        port: 9093
+        type: internal
+        tls: true
+        authentication:
+          type: custom
+          sasl: true
+          listener-config:
+            oauthbearer.sasl.client.callback.handler.class: client.class
+            oauthbearer.sasl.server.callback.handler.class: server.class
+            oauthbearer.sasl.login.callback.handler.class: login.class
+            oauthbearer.connections.max.reauth.ms: 999999999
+            sasl.enabled.mechanisms: oauthbearer
+            oauthbearer.sasl.jaas.config: |
+              org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required ;
+          secrets:
+            - name: example
+----
+
+.Protocol Map
+The protocol map that is generated for Strimzi uses the above `sasl` and `tls` fields to determine which protocol to map to the listener.
+
+* SASL = True, TLS = True -> SASL_SSL
+* SASL = False, TLS = True -> SSL
+* SASL = True, TLS = False -> SASL_PLAINTEXT
+* SASL = False, TLS = False -> PLAINTEXT
+
+.Listener Config
+All configuration listed under `listener-config` will be prefixed with `listener.name.<listener-name>-<port>`.
+Therefore, the above `sasl.enabled.mechanisms` becomes `listener.name.<listener-name>-<port>.sasl.enabled.mechanisms`.
+
+.Secrets
+`Secrets` will be mounted to `/opt/palantir/custom-auth-secrets/custom-<listener-name>-<port>/<secret-name>` in the Kafka broker nodes' containers.
+
+For example, the mounted secret `example` in the above setup would be located at `/opt/palantir/custom-auth-secrets/custom-oauth-bespoke-9093/example`.
```

Comment:
I don't think the paths correspond to the code (+ even if I read the code wrong, should be `/opt/kafka` ;-))

Reply:
woops! my apologies, silly mistake on my behalf :)